SourceX/DiabloUI/selhero.cpp

```diff
@@ -35,40 +35,40 @@ UI_Item SELHERO_DIALOG[] = {
+	{ { 265, 256, 320, 26 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), 0, listItems[0] },
+	{ { 265, 282, 320, 26 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), 1, listItems[1] },
+	{ { 265, 308, 320, 26 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), 2, listItems[2] },
+	{ { 265, 334, 320, 26 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), 3, listItems[3] },
+	{ { 265, 360, 320, 26 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), 4, listItems[4] },
+	{ { 265, 386, 320, 26 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), 5, listItems[5] },
+	{ { 239, 429, 120, 35 }, UI_BUTTON, (UiFlags)(UIS_CENTER | UIS_BIG | UIS_GOLD), 0, "OK", (void *)UiFocusNavigationSelect },
+	{ { 364, 429, 120, 35 }, UI_BUTTON, (UiFlags)(UIS_CENTER | UIS_BIG | UIS_DISABLED), 0, "Delete" },
+	{ { 489, 429, 120, 35 }, UI_BUTTON, (UiFlags)(UIS_CENTER | UIS_BIG | UIS_GOLD), 0, "Cancel", (void *)UiFocusNavigationEsc },
 };
 
 UI_Item SELCLASS_DIALOG[] = {
-	{ { 264, 211, 320, 33 }, UI_TEXT, UIS_CENTER | UIS_BIG, 0, "Choose Class" },
-	{ { 264, 285, 320, 33 }, UI_LIST, UIS_CENTER | UIS_MED | UIS_GOLD, UI_WARRIOR, "Warrior" },
-	{ { 264, 318, 320, 33 }, UI_LIST, UIS_CENTER | UIS_MED | UIS_GOLD, UI_ROGUE, "Rogue" },
-	{ { 264, 352, 320, 33 }, UI_LIST, UIS_CENTER | UIS_MED | UIS_GOLD, UI_SORCERER, "Sorcerer" },
-	{ { 279, 429, 140, 35 }, UI_BUTTON, UIS_CENTER | UIS_BIG | UIS_GOLD, 0, "OK", UiFocusNavigationSelect },
-	{ { 429, 429, 140, 35 }, UI_BUTTON, UIS_CENTER | UIS_BIG | UIS_GOLD, 0, "Cancel", UiFocusNavigationEsc },
+	{ { 264, 211, 320, 33 }, UI_TEXT, (UiFlags)(UIS_CENTER | UIS_BIG), 0, "Choose Class" },
+	{ { 264, 285, 320, 33 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), UI_WARRIOR, "Warrior" },
+	{ { 264, 318, 320, 33 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), UI_ROGUE, "Rogue" },
+	{ { 264, 352, 320, 33 }, UI_LIST, (UiFlags)(UIS_CENTER | UIS_MED | UIS_GOLD), UI_SORCERER, "Sorcerer" },
+	{ { 279, 429, 140, 35 }, UI_BUTTON, (UiFlags)(UIS_CENTER | UIS_BIG | UIS_GOLD), 0, "OK", (void *)UiFocusNavigationSelect },
+	{ { 429, 429, 140, 35 }, UI_BUTTON, (UiFlags)(UIS_CENTER | UIS_BIG | UIS_GOLD), 0, "Cancel", (void *)UiFocusNavigationEsc },
 };
 
 UI_Item ENTERNAME_DIALOG[] = {
-	{ { 264, 211, 320, 33 }, UI_TEXT, UIS_CENTER | UIS_BIG, 0, "Enter Name" },
-	{ { 265, 317, 320, 33 }, UI_EDIT, UIS_LIST | UIS_MED | UIS_GOLD, 15, heroInfo.name },
-	{ { 279, 429, 140, 35 }, UI_BUTTON, UIS_CENTER | UIS_BIG | UIS_GOLD, 0, "OK", UiFocusNavigationSelect },
-	{ { 429, 429, 140, 35 }, UI_BUTTON, UIS_CENTER | UIS_BIG | UIS_GOLD, 0, "Cancel", UiFocusNavigationEsc },
+	{ { 264, 211, 320, 33 }, UI_TEXT, (UiFlags)(UIS_CENTER | UIS_BIG), 0, "Enter Name" },
+	{ { 265, 317, 320, 33 }, UI_EDIT, (UiFlags)(UIS_LIST | UIS_MED | UIS_GOLD), 15, heroInfo.name },
+	{ { 279, 429, 140, 35 }, UI_BUTTON, (UiFlags)(UIS_CENTER | UIS_BIG | UIS_GOLD), 0, "OK", (void *)UiFocusNavigationSelect },
```

Comment:
I wonder if it would be better if I add an extra attribute (so that there will be one specific for UI_Art and one specific for click functions) in the struct to avoid the casting?